### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by ritikarawat220**
- Added integration specs using Capybara and Rspec
- Tested views: users/index, users/show, posts/index, and posts/view
- Checked if correct content is displayed on each page
- Checked if all links on the pages are working
- Fixed n+1 problem in the posts/index view

### Before
before

### After
after

### CODE REVIEW

File: spec/posts/index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  end
+
+  it 'I can see how many likes a post has.' do
+    page.has_content?(@post.likes_counter)
+  end
+
+  it 'I can see a section for pagination if there are more posts than fit on the view.' do
+    page.has_content?('Next')
+  end
+end
```
Comment (t-yanick) [inline spec/posts/index_spec.rb:52]: - [ ] Great job adding most User post index page test cases. At the moment, you are missing one test case: `Test that redirects to the post's show page when you click on a post.` Kindly add this test case
image

File: spec/posts/show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  end
+
+  it 'I can see the post body.' do
+    expect(page).to have_content(post.text)
+  end
+
+  it 'I can see the username of each commenter.' do
+    expect(page).to have_content(comment.author.name)
+  end
+end
```
Comment (t-yanick) [inline spec/posts/show_spec.rb:55]: - [ ]  Great job adding most User post show page test cases. At the moment, you are missing one test case: `Test that asserts you can see the comment each commentor left.`. Kindly add this test case
image

File: spec/users/show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    recent_posts.each do |post|
+      expect(page).to have_content(post.title)
+    end
+  end
+
+  it 'When I click to see all posts, it redirects me to the user\'s post index page.' do
+    click_link('See all posts')
+    expect(current_path).to eq(user_posts_path(@user))
+  end
+end
```
Comment (t-yanick) [inline spec/users/show_spec.rb:51]: - [ ]  Great job adding most User show page test cases. At the moment, you are missing one test case: `Test that redirects to the post's show page when you click on a user's post.`. Kindly add this test case.
image

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @ritikarawat220 , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good approach in solving N+1 problem
:heavy_check_mark: N+1 solution before and after screenshots added to PR body
:heavy_check_mark: Tests passing

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [ OPTIONAL] 
-  This is not a requirement for this milestone, but your validation test cases are failing.You can also consider fixing this.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Having a more descriptive branch name. `N+1` as branch name is not the best.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rotimiazeez — APPROVED** [review]
Hi @ritikarawat220 and @jeanbuhendwa 👏🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* Good job implementing the missing tests 👏🏾
* N+1 problem solved 🤝🏽
* All tests are passing ✅️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**